liquibase-service/src/main/resources/db/rdb_modern/routines/023-sp_d_interview_postprocessing-001.sql

```diff
@@ -0,0 +1,614 @@
+CREATE OR ALTER PROCEDURE dbo.sp_d_interview_postprocessing(
+    @interview_uids NVARCHAR(MAX),
+    @debug bit = 'false')
+as
+
+BEGIN
+
+    DECLARE @RowCount_no INT;
+    DECLARE @Proc_Step_no FLOAT = 0;
+    DECLARE @Proc_Step_Name VARCHAR(200) = '';
+    DECLARE @ColumnAdd_sql NVARCHAR(MAX) = '';
+    DECLARE @PivotColumns NVARCHAR(MAX) = '';
+    DECLARE @Pivot_sql NVARCHAR(MAX) = '';
+    DECLARE @Insert_sql NVARCHAR(MAX) = '';
+    DECLARE @Update_sql NVARCHAR(MAX) = '';
+    -- number of columns for the dynamic sql
+    DECLARE @Col_number BIGINT = (SELECT COUNT(*) FROM dbo.nrt_metadata_columns);
+
+    BEGIN TRY
+
+        SET @Proc_Step_no = 1;
+        SET @Proc_Step_Name = 'SP_Start';
+        DECLARE @batch_id bigint;
+        SET @batch_id = cast((format(GETDATE(), 'yyMMddHHmmss')) AS bigint);
+
+        BEGIN TRANSACTION;
+
+        SELECT @ROWCOUNT_NO = 0;
+
+        INSERT INTO [DBO].[JOB_FLOW_LOG]
+        (BATCH_ID, [DATAFLOW_NAME], [PACKAGE_NAME], [STATUS_TYPE], [STEP_NUMBER], [STEP_NAME], [ROW_COUNT])
+        VALUES (@BATCH_ID, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @PROC_STEP_NO, @PROC_STEP_NAME, @ROWCOUNT_NO);
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN TRANSACTION;
+
+        SET @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET @PROC_STEP_NAME = ' GENERATING #NEW_COLUMNS';
+
+        SELECT RDB_COLUMN_NM
+        INTO #NEW_COLUMNS
+        FROM dbo.nrt_metadata_columns
+        WHERE NEW_FLAG = 1
+          AND RDB_COLUMN_NM NOT IN (SELECT COLUMN_NAME
+                                    FROM INFORMATION_SCHEMA.COLUMNS
+                                    WHERE TABLE_NAME = 'D_INTERVIEW'
+                                      AND TABLE_SCHEMA = 'dbo');
+
+        SELECT @ROWCOUNT_NO = @@ROWCOUNT;
+
+        INSERT INTO [DBO].[JOB_FLOW_LOG]
+        (BATCH_ID, [DATAFLOW_NAME], [PACKAGE_NAME], [STATUS_TYPE], [STEP_NUMBER], [STEP_NAME], [ROW_COUNT])
+        VALUES (@BATCH_ID, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @PROC_STEP_NO, @PROC_STEP_NAME, @ROWCOUNT_NO);
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN TRANSACTION;
+
+        SET @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET @PROC_STEP_NAME = 'ADDING COLUMNS TO D_INTERVIEW';
+
+        SELECT @ColumnAdd_sql =
+               STRING_AGG('ALTER TABLE dbo.D_INTERVIEW ADD ' + QUOTENAME(RDB_COLUMN_NM) + ' VARCHAR(50);',
+                          CHAR(13) + CHAR(10))
+        FROM #NEW_COLUMNS;
+
+        IF @ColumnAdd_sql IS NOT NULL
+            BEGIN
+                EXEC sp_executesql @ColumnAdd_sql;
+            END
+
+        UPDATE dbo.nrt_metadata_columns
```

Comment:
Ooh an audit! Just had a thought that 1 usually implies an active flag in binary. 1 is new and 0 is implemented here?

Reply:
Yes, that is the correct understanding. If it's new, then it will be added if it doesn't already exist.

Reply:
Quick follow up, should the update statement exist within the IF condition or it makes more sense outside?